### ISSUE TITLE
Custom action in renderRowActions cannot access component state with isLoading State

### ISSUE THREAD

**Issue by CyrilB1531** [KevinVandy/material-react-table#335]
### material-react-table version

1.5.9

### react & react-dom versions

18.2.0

### Describe the bug and the steps to reproduce it

I'm using ReactMaterialTable with `State={ isLoading: data.length === 0 }` and a `button `in `renderRowActions`
I want to be able to get data array in clicked event of the action button but it is always empty.
If I add this line before the grid, all works perfectly (but no loading animation):
`if (data.length === 0) return <Fragment />`
 

### Minimal, Reproducible Example - (Optional, but Recommended)

```
import React, { useState, useEffect, useMemo, useCallback } from 'react'
import ReactMaterialTable, { MRT_ColumnDef, MRT_Row } from 'material-react-table'
import { IconButton } from '@mui/material'
import { OpenInBrowser } from '@mui/icons-material'
import { GetList } from '../../APICaller'

interface BookModel {
  gId: string
  name: string
  read: boolean
}

const BooksT = (): JSX.Element => {
  const [books, setBooks] = useState<BookModel[]>([])
  const [readBooks, setReadBooks] = useState<BookModel[]>([])
  const [dataCount, setDataCount] = useState(-1)
  const [loaded, setLoaded] = useState(false)

  useEffect(() => {
    if (!loaded) {
      setLoaded(true)
      return
    }
    GetList<BookModel>('/api/books')
      .then((datas) => {
        setBooks(datas)
        setDataCount(datas.length)
      })
      .catch((reason) => {
        console.error(reason)
        setDataCount(0)
      })
  }, [loaded])

  const columns = useMemo((): Array<MRT_ColumnDef<BookModel>> => {
    return [
      {
        accessorKey: 'gId',
        header: 'id',
        enableEditing: false
      },
      {
        accessorKey: 'name', // id used to define `group` column
        header: 'name'
      }
    ]
  }, [])

  const handleOnClick = useCallback(
    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, id: string): void => {
      event.preventDefault()
      fetch('/api/books/' + id + '/read', { method: 'PATCH' })
        .then((response) => {
          // something done with response
          const index = books.findIndex((a) => a.gId === id) // <== no data here
          books[index].read = true
          setReadBooks([...readBooks, books[index]])
          setBooks(books.slice(index, 1))
        })
        .catch(console.error)
    },
    [books, readBooks]
  )

  return (
    <ReactMaterialTable
      editingMode="row"
      enableBottomToolbar={false}
      enableDensityToggle={false}
      enableExpanding={false}
      enableFullScreenToggle={false}
      enablePagination={false}
      enableTopToolbar={false}
      enableColumnActions={false}
      enableEditing
      initialState={{ density: 'compact', columnVisibility: { gId: false } }}
      getRowId={(row) => row?.gId}
      columns={columns}
      data={books}
      state={{ isLoading: dataCount === -1 }}
      renderRowActions={({ row }: { row: MRT_Row<BookModel> }): JSX.Element => {
        console.log(books) // <== "Always empty array"
        return (
          <IconButton
            sx={{ color: 'inherit' }}
            onClick={(event) => {
              handleOnClick(event, row.original.gId)
            }}
            disabled={row.original.read}>
            <OpenInBrowser />
          </IconButton>
        )
      }}
    />
  )
}

export default BooksT

```

### Screenshots or Videos (Optional)

![screenshot 1](https://user-images.githubusercontent.com/47502913/214451530-9c0c7f0b-10a0-4f42-abd2-258bdc39ef38.jpg)

and with return <Fragment/> if no data loaded

![screenshot 2](https://user-images.githubusercontent.com/47502913/214452027-b1214496-bcbc-4270-a66b-7a49bd7e6abf.jpg)

### Do you intend to try to help solve this bug with your own PR?

No, because I do not have time to dig into it

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by CyrilB1531** [comment]
I have the same issue with release 1.5.10

**Comment by KevinVandy** [comment]
Would you be able to move this code to a sandbox so I can see what you mean?

**Comment by CyrilB1531** [comment]
of course. this is the link: [https://playcode.io/1099471](https://playcode.io/1099471)

**Comment by Kliton** [comment]
Same issue: i cannot access component state in renderRowActions

**Comment by arashn0uri** [comment]
Are there any updates on this issue?

**Comment by KevinVandy** [comment]
I can definitely reproduce the issue, but have not found a solution or fix yet. A work around is to not render the table until the data for the row actions has loaded, or to add a key prop that you pass to the table when the data for the row actions changes.

**Comment by tdanh2k** [comment]
I think React Context should be a temporary workaround for this problem.

**Comment by KevinVandy** [comment]
@tdanh2k @arashn0uri @Kliton @CyrilB1531 Better late than never. This is fixed in v1.9.4

https://www.material-react-table.dev/?path=/story/fixed-bugs-useeffects--render-row-actions-async
https://www.material-react-table.dev/?path=/story/fixed-bugs-useeffects--render-row-action-menu-items-async
